recipes/newrelic/elasticsearch/debian.yml

```diff
@@ -0,0 +1,137 @@
+    prompt: "Please enter your ElasticSearch username used to connect to API."
+  - name: "NR_CLI_API_HOSTNAME"
+    prompt: "Please enter your ElasticSearch hostname below. If none is provided then the default value: localhost, will be used."
+    default: "localhost"
+  - name: "NR_CLI_API_PORT"
+    prompt: "Please enter your ElasticSearch port below. If none is provided, the default value: 9200, will be used."
+    default: 9200
+  - name: "NR_CLI_CONFIG_PATH"
+    prompt: "Please provide the path to ElasticSearch config file, if none is provided then the default: /etc/elasticsearch/elasticsearch.yml will be used."
+    default: "/etc/elasticsearch/elasticsearch.yml"
+  - name: "NR_CLI_API_USE_SSL"
+    prompt: "Do you want to use SSL to connect? (true/false)"
+    default: false
+  - name: "NR_CLI_API_CA_BUNDLE_DIR"
+    prompt: "Please enter the directory in which the SSL certificate is located on the host. If using SSL please leave this field empty."
+  - name: "NR_CLI_API_CA_BUNDLE_FILE"
+    prompt: "Please enter the path to the SSL certificate file on the host. If using SSL please leave this field empty."
+
+install:
+  version: "3"
+  silent: true
+
+  tasks:
+    default:
+      cmds:
+        - taks: assert_pre_req
+        - task: setup
+
+    assert_pre_req:
+      cmds:
+        - |
+          SERVICE_EXIST=$(sudo systemctl status newrelic-infra.service | grep "Active" | wc -l)
+          if [ $SERVICE_EXIST -eq 0 ]; then
+            echo "The newrelic-infra agent service is NOT installed on the host, but is required to install this integration." >> /dev/stderr
+            exit 1
+          fi
+
+    setup:
+      label: "Installing elasticsearch integration..."
+      cmds:
+        - |
+          sudo mkdir -p "/etc/newrelic-infra/integrations.d"
+        - |
+          sudo apt-get update
+        - |
+          sudo apt-get install nri-elasticsearch -y
+        - |
+          if [ -f /etc/newrelic-infra/integrations.d/elasticsearch-config.yml ]; then
+            sudo rm /etc/newrelic-infra/integrations.d/elasticsearch-config.yml;
+          fi
+
+          sudo touch /etc/newrelic-infra/integrations.d/elasticsearch-config.yml;
+        - |
+          if [ "$NR_CLI_API_USE_SSL" == false ]; then
+            sudo tee -a /etc/newrelic-infra/integrations.d/elasticsearch-config.yml > /dev/null <<"EOT"
+          integration_name: com.newrelic.elasticsearch
+
+          instances:
+            - name: elasticsearch
+              command: all
+              arguments:
+                cluster_environment: staging
+                config_path: {{.NR_CLI_API_CONFIG_PATH}}
+                hostname: {{.NR_CLI_API_HOSTNAME}}
+                local_hostname: localhost
+                username: {{.NR_CLI_API_USERNAME}}
+                password: {{.NR_CLI_API_PASSWORD}}
+                port: {{.NR_CLI_API_PORT}}
+                timeout: 30
+                use_ssl: false
+                ca_bundle_dir: 
```

Comment:
@JakubKotkowiak similarly, we should be using the NR_CLI_API_CA_BUNDLE_DIR and NR_CLI_API_CA_BUNDLE_FILE input param here.

Reply:
Not really, it's done this way to prevent the tests from hanging. If we put empty values as those two fields then it hangs up since the deployer treats it as if those fields didn't exist. In this case we aren't using ssl so those two fields are left empty/

Reply:
@JakubKotkowiak after reviewing the implementation of the Elastic Search OHI at https://github.com/newrelic/nri-elasticsearch
It appears the use_ssl flag seems to not be used.
Consequently, perhaps we could simplify this recipe implementation to:
- if SSL is not used, write a config without any SSL configuration (to avoid any ambiguity).
- if SSL is used, keep the config you have below.

In the test context, you can then set the env vars to:
- use_ssl false
- ca bundle "not used"
- ca file "not used"
And those values would actually not be written to the actual config file.

Reply:
Sure thing, that's what we will try then. Thanks for the feedback